### ISSUE TITLE
import from system-address book aborts on erroneous email-addresses

### ISSUE THREAD

**Issue by emulti** [deltachat/deltachat-android#1201]
- Android version: 8.1.0
- Device: Wileyfox Swift 2 X
- Delta Chat version: 1.0.3
- Expected behavior: deltachat searches Android contacts when starting new chat
- Actual behavior: dc only searches list of existing chats
- Steps to reproduce the problem: click '+' for start new chat, start typing in 'Enter name or email address' field to search
- Screenshots:
- Logs:

Maybe I have misunderstood the purpose of the 'Contacts' permission and this is intended behaviour? I was expecting a search in 'New chat' to allow typing (part of) a contact's name, producing a list of hits with email addresses, one of which can be selected for the target of the chat session. However, it seems that the search scope is only across existing chats. 

Android contacts book is synced from Nextcloud 17 via DavXs.  Other applications can access/search the contacts book. Deselecting/reselecting the Contacts permission does not change the behaviour.

**Comment by r10s** [comment]
if you give Delta Chat permission to access the address book, all email-addresses will be copied from the system's address book to Delta Chat's contact list and are accessible from there.

therefore, you should be able to search for all email addresses and the belonging names.

**Comment by r10s** [comment]
in other words: not all contacts from the system address book will appear in Delta Chat - eg. contacts without email-addresses won't be there.

@emulti please close this issue if this is the case on your system and things work as described.

**Comment by emulti** [comment]
Thanks for responding! 
On this version combination at least, that doesn't appear to happen. Delta Chat's contact list is empty untill new contacts are manually created. 
Is there a wayto trigger a 're-sync'?

**Comment by r10s** [comment]
> Is there a wayto trigger a 're-sync'?

it should be done automatically whenever you open the contact list.

can you please check if the system shows that Delta Chat has really access to the system's address book.

**Comment by emulti** [comment]
Checked again, and permission to access contacts is definitely 'on' in the Settings/Applications/permissions view of Android. 
Android 8.1.0
Deltachat 1.0.3 from Google Play

**Comment by r10s** [comment]
ok, thanks for rechecking and confirming.

**Comment by emulti** [comment]
Found the cause in the log. Once contact had a mobile number (numeric-only string) in one of the email address fields. 
After a restart of Deltachat, some (not all) contacts were imported. Unfortunately, because of the restart the log contianing the exact error was overwritten, but the message was from contact.rs with a form like 'invalid address 07123456789 in contact "John Doe"', and no contacts were imported.

Now, the import process ended with the message: 
    W DeltaChat: ThreadId(19) src/contact.rs:328: Bad address "johndoe@domain" for contact "John Doe".
After correcting this data, the import process continued (alphabetically) to completion. 
So it seems that if there are any malformed email addresses in the contact list, contact import stops at that record.
Does this help you to debug?

**Comment by r10s** [comment]
> Does this help you to debug?

yip, great finding!

**Comment by r10s** [comment]
@Hocuri maybe you can have a look at that, if you find the time, seems to be related to the things you've recently working on, so you are a bit in this topic.

**Comment by Hocuri** [comment]
I failed to reproduce this and I did not find any possible cause in the code. 

When I tried to reproduce this, I got two warnings (not errors!) (yes, I checked it, the core sends two warnings for one erroneous addr, maybe this should be changed)
```
2020-05-16 16:57:48.520 8894-10321/com.b44t.messenger.beta W/DeltaChat: ThreadId(23) src/contact.rs:346: Bad address "849452" for contact "Dbg".
2020-05-16 16:57:48.520 8894-10321/com.b44t.messenger.beta W/DeltaChat: ThreadId(23) src/contact.rs:497: Failed to add address 849452 from address book: Bad address supplied: "849452"
```
but the other contacts were imported. Also, [the core only sends a warning when it gets a bad address and then continues with the next address](https://github.com/deltachat/deltachat-core-rust/blob/fff1eaba45110833f4e7d371b2eb254ef6e3a3a7/src/contact.rs#L497).

Seems that the problem was something else.
